### PR TITLE
mail-filter/rspamd: version bump and sync live

### PR DESCRIPTION
**Pull request by arkamar**
I have bumped rspamd to 3.2 and synced live ebuild with it. It was possible to unbundle `dev-lua/argparse`. `pcre2` is enabled by default in upstream, so I have changed it to `+pcre2`.
The `jemalloc` bug is still present and I don't think it will be fixed anytime soon, therefore I have changed the mask to `>=mail-filter/rspamd-3.0`.

### CODE REVIEW

File: mail-filter/rspamd/rspamd-3.2.ebuild [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+	sys-apps/file
+	blas? (
+		virtual/blas
+		virtual/lapack
+	)
+	cpu_flags_x86_ssse3? ( dev-libs/hyperscan )
+	jemalloc? ( dev-libs/jemalloc:= )
+	dev-libs/openssl:0=[-bindist(-)]
+	pcre2? ( dev-libs/libpcre2:=[jit=] )
+	!pcre2? ( dev-libs/libpcre[jit=] )
```
Comment (thesamesam) [inline mail-filter/rspamd/rspamd-3.2.ebuild:53]: Honestly, at this point, I wonder if we should just be dropping pcre and unconditionally using pcre2. But not a big deal. We can do it on next release. Making it default-on is a step in the right direction.

File: mail-filter/rspamd/rspamd-3.2.ebuild [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+"
+DEPEND="${RDEPEND}
+	<dev-cpp/doctest-2.4.7
```
Comment (thesamesam) [inline mail-filter/rspamd/rspamd-3.2.ebuild:56]: This < dep still needed? Can you add a bug reference for it?

Comment (arkamar) [reply mail-filter/rspamd/rspamd-3.2.ebuild:56]: yes, it is still needed, I wrote it to the commit message, but it makes sense to comment it here as well, I'll fix it.

File: mail-filter/rspamd/rspamd-3.2.ebuild (hunk 1 above)
Comment (arkamar) [reply mail-filter/rspamd/rspamd-3.2.ebuild]: I dropped `pcre2` useflag together with `dev-libs/libpcre` dependency, it is eol anyway.

**Pull request closed without merging**